src/main/java/seedu/rc4hdb/logic/parser/commandparsers/ColumnManipulatorCommandParser.java

```diff
@@ -47,28 +48,17 @@ private static HashMap<String, List<String>> validateAndGenerateFieldLists(Strin
 
         assert(lettersOfColumnsToShowOrHide != null);
 
-        // Non-existent mappings will result in null values in the final list... null checks are performed below
-        List<String> fullNamesOfColumnsToShowOrHide = lettersOfColumnsToShowOrHide.stream()
-                .map(LETTER_TO_FIELD_NAME_MAPPINGS::get)
-                .collect(Collectors.toList());
+        // Non-existent mappings will result in null values in the final list...
+        // Checks for these null values are performed in the populateBaseAndComplementLists method below
+        List<String> fullNamesOfColumnsToShowOrHide = getFullNamesOfColumnsToShowOrHide(lettersOfColumnsToShowOrHide);
 
         List<String> baseList = new ArrayList<>();
         List<String> complementList = new ArrayList<>(ColumnManipulatorCommand.ALL_FIELDS);
-
-        for (String column : fullNamesOfColumnsToShowOrHide) {
-            if (column == null) {
-                throw new ParseException(INVALID_FIELDS_ENTERED);
-            }
-            if (!ColumnManipulatorCommand.ALL_FIELDS.contains(column)) {
-                throw new ParseException(INVALID_FIELDS_ENTERED);
-            }
-            baseList.add(column);
-            complementList.remove(column);
-        }
+        populateBaseAndComplementListsWithColumns(fullNamesOfColumnsToShowOrHide, baseList, complementList);
 
         HashMap<String, List<String>> hashMap = new HashMap<>();
-        hashMap.put(BASE_LIST, baseList);
-        hashMap.put(COMPLEMENT_LIST, complementList);
+        populateHashMapWithBaseAndComplementLists(hashMap, baseList, complementList);
```

Comment:
The creation of the `HashMap` can be done within the `populateHashMapWithBaseAndComplementLists` method. This will make the code cleaner, while removing redundant code.

Reply:
Hi Jordan, I have decided to preserve the `populateHashMapWithBaseAndComplementLists` method for consistency with the other `populateBaseAndComplementListsWithColumns` method.

Reply:
This would also reduce the need to instantiate another variable to hold the `HashMap` instance before we return it.